storage/aptosdb/src/state_merkle_db.rs

```diff
@@ -166,7 +168,15 @@ impl StateMerkleDb {
                 .collect::<Vec<_>>()
                 .par_iter()
                 .with_min_len(128)
-                .map(|row| batch.put::<StaleNodeIndexSchema>(row, &()))
+                .map(|row| {
+                    if previous_epoch_ending_version.is_some()
+                        && row.node_key.version() <= previous_epoch_ending_version.unwrap()
+                    {
+                        batch.put::<StaleNodeIndexCrossEpochSchema>(row, &())
```

Comment:
Would it be better to change the `stale_since_version` to be the `previous_epoch_ending_version`?

Reply:
discussed elsewhere that we want them to go away gradually.